src/DynamoCore/Search/NodeSearchModel.cs

```diff
@@ -48,58 +50,108 @@ internal XmlDocument ComposeXmlForLibrary()
+                String.Join(",", entry.SearchKeywords.Where(word => !String.IsNullOrWhiteSpace(word))));
+
+            var dynamoNode = entry.CreateNode();
+
             // If entry has input parameters.
-            if (entry.InputParameters.First().Item2 != Properties.Resources.NoneString)
+            if (dynamoNode.InPorts.Count != 0)
             {
                 var inputNode = XmlHelper.AddNode(element, "Inputs");
-                foreach (var parameter in entry.InputParameters)
+                for (int i = 0; i < dynamoNode.InPorts.Count; i++)
                 {
                     var parameterNode = XmlHelper.AddNode(inputNode, "InputParameter");
-                    XmlHelper.AddAttribute(parameterNode, "Name", parameter.Item1);
-                    XmlHelper.AddAttribute(parameterNode, "Type", parameter.Item2);
+                    XmlHelper.AddAttribute(parameterNode, "Name", dynamoNode.InPorts[i].PortName);
+
+                    // Case for UI nodes as ColorRange.
+                    if (dynamoNode.InPorts.Count == entry.InputParameters.Count()
+                        && entry.InputParameters.First().Item2 != Properties.Resources.NoneString)
+                    {
+                        XmlHelper.AddAttribute(parameterNode, "Type", entry.InputParameters.ElementAt(i).Item2);
+                    }
+
+                    // Add default value, if it's not null.
+                    if (dynamoNode.InPorts[i].DefaultValue != null)
+                    {
+                        XmlHelper.AddAttribute(parameterNode, "DefaultValue",
+                            dynamoNode.InPorts[i].DefaultValue.ToString());
+                    }
+                    XmlHelper.AddAttribute(parameterNode, "Tooltip", dynamoNode.InPorts[i].ToolTipContent);
                 }
             }
 
             // If entry has output parameters.
-            if (entry.OutputParameters.First() != Properties.Resources.NoneString)
+            if (dynamoNode.OutPorts.Count != 0)
             {
-                var inputNode = XmlHelper.AddNode(element, "Outputs");
-                foreach (var parameter in entry.OutputParameters)
+                var outputNode = XmlHelper.AddNode(element, "Outputs");
+                for (int i = 0; i < dynamoNode.OutPorts.Count; i++)
                 {
-                    var parameterNode = XmlHelper.AddNode(inputNode, "OutputParameter");
-                    XmlHelper.AddAttribute(parameterNode, "Type", parameter);
+                    var parameterNode = XmlHelper.AddNode(outputNode, "OutputParameter");
+
+                    // Case for UI nodes as ColorRange.
+                    if (dynamoNode.OutPorts.Count == entry.OutputParameters.Count()
+                        && entry.OutputParameters.First() != Properties.Resources.NoneString)
+                    {
+                        XmlHelper.AddAttribute(parameterNode, "Type", entry.OutputParameters.ElementAt(i));
```

Comment:
Color.Components out ports are set as [MultiReturn("a", "r", "g", "b")].
We didn't add types for `MultiReturn`:( Maybe, we should create additional task for this?
Even in Dynamo it's not said what type port returns.... `var` is tooltip by default.
http://prntscr.com/asi61m